src/components/html-injection/ko/widgetViewModelBinder.ts

```diff
@@ -0,0 +1,88 @@
+import { Bag } from "@paperbits/common";
+import { EventManager, Events } from "@paperbits/common/events";
+import { IWidgetBinding } from "@paperbits/common/editing";
+import { widgetName, widgetDisplayName, widgetEditorSelector } from "../constants";
+import { WidgetViewModel } from "./widgetViewModel";
+import { ViewModelBinder } from "@paperbits/common/widgets";
+import { StyleCompiler } from "@paperbits/common/styles";
+import { JssCompiler } from "@paperbits/styles/jssCompiler";
+import { ISettingsProvider } from "@paperbits/common/configuration";
+import { HTMLInjectionWidgetModel } from "../widgetModel";
+
+/**
+ * This class describes how the model needs to be presented (as a view model)
+ * in a specific UI framework.
+ */
+export class WidgetViewModelBinder implements ViewModelBinder<HTMLInjectionWidgetModel, WidgetViewModel>  {
+    constructor(
+        private readonly eventManager: EventManager,
+        private readonly styleCompiler: StyleCompiler,
+        private readonly settingsProvider: ISettingsProvider,
+    ) { }
+
+    public async updateViewModel(model: HTMLInjectionWidgetModel, viewModel: WidgetViewModel): Promise<void> {
+        let htmlStyling: string = "";
+
+        if (model.inheritStyling) {
+            const environment = await this.settingsProvider.getSetting<string>("environment");
+
+            htmlStyling = '<link href="/styles/theme.css" rel="stylesheet" type="text/css">';
+
+            if (environment === "development") {
+                // Access global style sheet
+                const globalStyleSheet = await this.styleCompiler.getStyleSheet();
+
+                // Compile to CSS
+                const compiler = new JssCompiler();
+                htmlStyling += `<style>${compiler.compile(globalStyleSheet)}</style>`;
+
+                // Reflect style changes in designer
+                // TODO: looks like this is not needed after all
+                /* this.eventManager.addEventListener("onStyleChange", async (styleSheet) => {
+                    console.log({styleSheet});
+                    // viewModel.htmlStyling(styleSheet);
+                    // await this.updateViewModel(model, viewModel);
+                    // this.eventManager.dispatchEvent(Events.ContentUpdate);
+                }); */
```

Comment:
@azaslonov it appears it works just fine even without this event listener, changes from the Styles page are propagated correctly - I guess that was this event listener supposed to handle